api/host.go

```diff
@@ -166,40 +167,109 @@ func (api *API) hostHandlerPOST(w http.ResponseWriter, req *http.Request, _ http
 		var x types.Currency
 		_, err := fmt.Sscan(req.FormValue("mincontractprice"), &x)
 		if err != nil {
-			WriteError(w, Error{"Malformed mincontractprice"}, http.StatusBadRequest)
-			return
+			return modules.HostInternalSettings{}, nil
 		}
 		settings.MinContractPrice = x
 	}
 	if req.FormValue("mindownloadbandwidthprice") != "" {
 		var x types.Currency
 		_, err := fmt.Sscan(req.FormValue("mindownloadbandwidthprice"), &x)
 		if err != nil {
-			WriteError(w, Error{"Malformed mindownloadbandwidthprice"}, http.StatusBadRequest)
-			return
+			return modules.HostInternalSettings{}, nil
 		}
 		settings.MinDownloadBandwidthPrice = x
 	}
 	if req.FormValue("minstorageprice") != "" {
 		var x types.Currency
 		_, err := fmt.Sscan(req.FormValue("minstorageprice"), &x)
 		if err != nil {
-			WriteError(w, Error{"Malformed minstorageprice"}, http.StatusBadRequest)
-			return
+			return modules.HostInternalSettings{}, nil
 		}
 		settings.MinStoragePrice = x
 	}
 	if req.FormValue("minuploadbandwidthprice") != "" {
 		var x types.Currency
 		_, err := fmt.Sscan(req.FormValue("minuploadbandwidthprice"), &x)
 		if err != nil {
-			WriteError(w, Error{"Malformed minuploadbandwidthprice"}, http.StatusBadRequest)
-			return
+			return modules.HostInternalSettings{}, nil
 		}
 		settings.MinUploadBandwidthPrice = x
 	}
 
-	err := api.host.SetInternalSettings(settings)
+	return settings, nil
+}
+
+// hostEstimateScoreGET handles the POST request to /host/estimatescore and
+// computes an estimated HostDB score for the provided settings.
+func (api *API) hostEstimateScorePOST(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	entry, exists := api.renter.Host(api.host.PublicKey())
```

Comment:
yeah, I think uptime should be ignored, since it can't be accurately measured. But then how do we estimate the ranking? Perhaps the uptime of every host should be set to some sane value before their score is calculated.